client.go

```diff
@@ -107,6 +108,10 @@ func (client Client) dispatchRequest(request *http.Request) ([]byte, error) {
 func (client Client) dispatchSingleRequest(request *http.Request) ([]byte, error) {
 	client.Signer.OAuthSign(request)
 	httpClient := http.Client{}
+	if client.HTTPClient != nil {
+		httpClient = *client.HTTPClient
```

Comment:
This works because all the fields of `http.Client` are currently pointers, but I agree that this behaviour may not be guaranteed in the future. I've changed this to:
```golang
httpClient := &http.Client{}
if client.HTTPClient != nil {
	httpClient = client.HTTPClient
}
```